File: practice_java_program/notpad/HappyNumber.java
public class HappyNumber{
	public static void main(String[] args){
		int num=19;
		
		int fast = num;
		int slow = num;
		while(true){
			slow=getSum(slow); //by 1 step
			fast = getSum(getSum(fast)); //2 step
			if(slow==fast){
				break;
			}
		}
		if (slow==1){
			System.out.println("this is happy num");
		}
		else{
			System.out.println("this is not a happy num");
		}	
	}
	public static int getSum(int num){
		int sum=0;
		while(num>0){
			int digit=num%10;
			sum += digit*digit;
			num=num/10;
		}
		return sum;
	}
}

File: python_oriented_interview/notpad/HappyNumber.py
def get_sum(num):
    sum_val = 0
    while num > 0:
        digit = num % 10
        sum_val += digit * digit
        num = num // 10
    return sum_val

def is_happy(num):
    fast = num
    slow = num
    while True:
        slow = get_sum(slow)
        fast = get_sum(get_sum(fast))
        if slow == fast:
            break
    return slow == 1

num = 19
if is_happy(num):
    print("this is happy num")
else:
    print("this is not a happy num")
